Final Touchups on Code

jeremy23newman opened the pull request:
Tweaking code to fit checkstyle, and ensuring it passes gradle checks.

DijonnaWatson committed:
Update WriteResult.java

making this status checker match the factorComputeResult interface

DijonnaWatson committed:
Update FactorComputeEngine.java

I just realized I don't think there's a reason for the delimiter to be a method here since it's already defined in the FactorComputeRequest.

DijonnaWatson committed:
Add files via upload

DijonnaWatson committed:
Delete src/factorFinder/FactorComputeEngineImpl.class

wrong file

DijonnaWatson committed:
Update FactorComputeEngine.java

DijonnaWatson committed:
Add files via upload

this was the file i meant to upload earlier

DijonnaWatson committed:
Update FactorComputeEngineImpl.java

what i got so far for the compute engine implementation. I'm also just fixing the indentation

DijonnaWatson committed:
Update FactorComputeEngineImpl.java

ViceRaven committed:
added delimiter logic to FactorComputeEngineImpl.java

ViceRaven committed:
Create FactorComputationCoordinationImpl.java

ViceRaven committed:
added computation logic to FactorComputationCoordinationImpl.java

ViceRaven committed:
forgot to add list import InputConfig.java

DijonnaWatson committed:
Update WriteResult.java

nvm i changed it back to the original since it messes with certain things already using it in other classes

ViceRaven committed:
Revamped FactorComputationCoordinationImpl.java

Original file was way to confusing and didn't have the steps for 2 explicitly outline. Hopefully this revamped version will give more peace of mind and help us follow along coherently.

DijonnaWatson committed:
Update FactorComputeEngineImpl.java

added the default delimiter back

DijonnaWatson committed:
Update FactorComputeEngineImpl.java

ViceRaven committed:
Create FactorDataStoreImpl.java

DijonnaWatson committed:
Update and rename FactorComputationCoordinationImpl.java to FactorComputationCoordinatorImpl.java

fixing package name, and outter file name

DijonnaWatson committed:
Update WriteResult.java

fixed the WriteResult again to match the FactorComputeResult

ViceRaven committed:
Update FactorDataStoreImpl.java with code from eclipse

Retrieves input data from input config
Also contains if-else for when the data is null or not

ViceRaven committed:
Update FactorComputationCordinatorPrototype.java

small change to print out a better confirmation message upon getstatus

ViceRaven committed:
Updated package on WriteResult.java

ViceRaven committed:
added missing import statements FactorComputeRequest.java

ViceRaven committed:
added missing package Update OutputConfig.java

ViceRaven committed:
Update FactorDataStorePrototype.java

had to tweak prototype to work with the coordination implementation

added some things like output configuration, processing loaded data, and checking the write result status.

ViceRaven committed:
Updated FactorComputationCordinatorPrototype.java

added method stub from eclipse

ViceRaven committed:
Updated FactorDataStore.java

added missing "List<String> list" from eclipse

ViceRaven committed:
Updated OutputConfig.java

output config was missing list import

ViceRaven committed:
Update FactorDataStorePrototype.java

deleted eclipse package and re-run through code formatter to fix checkstyle

ViceRaven committed:
Update FactorDataStorePrototype.java

another attempt to fix checkstyle

jeremy23newman committed:
reverting gradle.yml(step 6)

jeremy23newman committed:
adding import statement for java.util.list

jeremy23newman committed:
adding import statement for java.util.list

jeremy23newman committed:
removing imports

jeremy23newman committed:
adding reformatted import statements

DijonnaWatson committed:
Update TestFactorDataStore.java

updating the test

DijonnaWatson committed:
Update TestFactorComputeEngine.java

updating this test as well

jeremy23newman committed:
adding test for negative numbers

DijonnaWatson committed:
Update TestFactorComputeEngine.java

fixing the imports

ViceRaven committed:
Update FactorDataStoreImpl.java

added file reading and writing functionality to fulfill the requirements of step 3 (although a bit iffy on the part where only the tests from assign 3 will be using it. might have gone overboard)

ViceRaven committed:
Added the file input and output config files

Added the implementations of our input and output config

ViceRaven committed:
Updated InputConfig.java

just a method to get the file path for input data

ViceRaven committed:
Updated OutputConfig.java

just a method to get the file path for output data

ViceRaven committed:
UpdateD FactorDataStoreImpl.java

removed the .io* import because the checkstyle hates imports like that for whatever reason

ViceRaven committed:
Update FactorDataStorePrototype.java

deleted unnecessary @override

ViceRaven committed:
Update FactorComputationCordinatorPrototype.java

deleted unnecessary @override

ViceRaven committed:
Update FactorComputationCordinatorPrototype.java

added missing getfilepath constructor and for now just had it return an empty string until we test.

ViceRaven committed:
Update FactorDataStorePrototype.java

same change done to factor comp cord protoype added missing getfilepath constructor and for now it just returns an empty string

ViceRaven committed:
Update InMemoryFactorDataStore.java

added missing import

ViceRaven committed:
Update InMemoryInputConfig.java

added missing imports

ViceRaven committed:
Update InMemoryOutputConfig.java

added missing inputs

ViceRaven committed:
Update InMemoryInputConfig.java

added missing getFilePath method

ViceRaven committed:
Update InMemoryOutputConfig.java

added missing getFilePath method

ViceRaven committed:
Update InMemoryInputConfig.java

Fixed getfilepath

ViceRaven committed:
Update InMemoryOutputConfig.java

fixed getfilepath

ViceRaven committed:
Update InMemoryOutputConfig.java

fixed input config being declared in output config

DijonnaWatson committed:
Update TestFactorDataStore.java

updating the testdataStore

DijonnaWatson committed:
Update TestFactorComputeEngine.java

updating the test engine

DijonnaWatson committed:
Update TestFactorComputeEngine.java

fix indentation

DijonnaWatson committed:
Update TestFactorDataStore.java

fixing some comments

ViceRaven committed:
Update TestComputeEngineIntegration.java

another attempt to fix test files by adding missing imports

ViceRaven committed:
Update TestComputeEngineIntegration.java

removed imports because they did not work

ViceRaven committed:
Update InMemoryFactorDataStore.java

added missing implementation

ViceRaven committed:
Update TestComputeEngineIntegration.java

attempting to fix by directly implementing factor computer engine impl

ViceRaven committed:
Update TestComputeEngineIntegration.java

another attempt to fix

ViceRaven committed:
Update TestComputeEngineIntegration.java

added another fix

ViceRaven committed:
Update TestFactorDataStore.java

fixed wrong package name

ViceRaven committed:
Update TestFactorComputationCordinator.java

fixed missing package

DijonnaWatson committed:
Update TestFactorComputeEngine.java

fixing package

jeremy23newman committed:
adding import statements

jeremy23newman committed:
removing imports

ViceRaven committed:
Update InMemoryFactorDataStore.java

initialized stored results

ViceRaven committed:
Update InMemoryFactorDataStore.java

forgot array list impl

jeremy23newman committed:
Update TestComputeEngineIntegration.java

ViceRaven committed:
Update TestFactorComputationCordinator.java

overrode abstract method of getfilepath

jeremy23newman committed:
Update TestComputeEngineIntegration.java

jeremy23newman committed:
Update TestComputeEngineIntegration.java

ViceRaven committed:
Update InputConfig.java

added missing override

ViceRaven committed:
Update TestFactorComputationCordinator.java

removed override

ViceRaven committed:
Update InputConfig.java

removed override

ViceRaven committed:
Update TestFactorComputationCordinator.java

tried new override position

ViceRaven committed:
Update TestFactorComputationCordinator.java

added missing getinputdata method from input config

jeremy23newman committed:
attempting to fix integration test

jeremy23newman committed:
adding import statements

ViceRaven committed:
Update TestFactorComputationCordinator.java

corrected the implementation of the getinputdata method

ViceRaven committed:
Update TestFactorComputationCordinator.java

added missing imports

ViceRaven committed:
Update TestFactorComputationCordinator.java

missing semi colon

DijonnaWatson committed:
Update TestFactorDataStore.java

fixing some comments since we had updated some code based off the implementations we did

jeremy23newman committed:
updating to fix checkstyle

jeremy23newman committed:
Update TestComputeEngineIntegration.java

ViceRaven committed:
Updated FactorComputationCoordinatorImpl.java

Cleaned up comments to help with readability by directly correlating blocks of code to the functional requirements (this can be changed by you guys if you think you can make it flow better).

DijonnaWatson committed:
Update TestComputeEngineIntegration.java

cleaning up some code

DijonnaWatson committed:
Update FactorDataStoreImpl.java

fixing some comments to be more descriptive

jeremy23newman committed:
Update TestFactorDataStore.java

DijonnaWatson committed:
Update FactorDataStorePrototype.java

added some notes to remind us that there are already implementations of these instances

DijonnaWatson committed:
Update InMemoryInputConfig.java

coments

jeremy23newman committed:
Update TestFactorComputeEngine.java

ViceRaven committed:
Update FactorComputationCordinatorPrototype.java

Revamped some comments for clarity and readability also did a minor change with our getter methods

DijonnaWatson committed:
Update FileOutputConfig.java

fixing some todo comments

DijonnaWatson committed:
Update FactorComputeRequest.java

adding my comments back

DijonnaWatson committed:
Update FileOutputConfig.java

deleted an extra method that we didn't need

DijonnaWatson committed:
Update OutputConfig.java

fixing up the methods, and got rid of an extra method we didn't need

jeremy23newman committed:
fixing outputconfig in test

ViceRaven commented on src/factorFinder/OutputConfig.java:
```diff
@@ -1,4 +1,7 @@
+
 public interface OutputConfig {
   void writeOutputData(List<String> data);
-  List<String> getOutputData();
```
Oh forgot to delete that, thanks!

DijonnaWatson commented on src/factorFinder/FactorComputeRequest.java:
```diff
@@ -1,4 +1,7 @@
+
 package com.example.factorfinder;
+import com.example.factorfinder.InputConfig;
```
It's good that you guys figured out these specified imports were one of the reasons why the gradle wrapper kept saying it could find the specific files.

ViceRaven committed:
Update TestFactorComputationCordinator.java

attempt to fix checkstyle errors.

ViceRaven committed:
Update FactorComputationCordinatorPrototype.java

attempted to fix gradle by adding missing arguments

jeremy23newman committed:
Update FactorComputationCordinatorPrototype.java

jeremy23newman committed:
Update FactorComputationCordinatorPrototype.java

jeremy23newman committed:
Update TestComputeEngineIntegration.java

jeremy23newman committed:
Update TestComputeEngineIntegration.java

ViceRaven committed:
Update TestComputeEngineIntegration.java

fixed extra end parentheses

ViceRaven committed:
Update TestComputeEngineIntegration.java

added back parentheses

ViceRaven committed:
Update TestComputeEngineIntegration.java

removed extra parenth

jeremy23newman committed:
adding import

jeremy23newman committed:
removing imports

ViceRaven committed:
Update TestComputeEngineIntegration.java

declare output config

ViceRaven committed:
Update TestComputeEngineIntegration.java

fixed error converting factors into string list. must be using intstream and .maptoobject instead

ViceRaven committed:
Update TestComputeEngineIntegration.java

added missing import

ViceRaven committed:
Update TestComputeEngineIntegration.java

added missing import

ViceRaven committed:
Update InMemoryOutputConfig.java

attempted to fix gradle by removing uneeded method

ViceRaven committed:
Update FactorComputeEngineImpl.java

the logic to get the factors was causing the gradle to fail. specifically where we calculate the factors of 10. there doesn't seem to be any errors with this new logic in eclipse but the gradle will decide if we have to edit anything

ViceRaven committed:
Update TestFactorComputationCordinator.java

changed output config to use a mock version because null may have been giving a null pointer error

ViceRaven committed:
Update TestFactorComputationCordinator.java

forgot to change the inputconfig and outputconfig to mock versions when creating compute request

ViceRaven committed:
Update TestFactorComputationCordinator.java

testing leaving inputconfig without mock to satisfy gradle

DijonnaWatson committed:
Update FactorComputeEngineImpl.java

fixing indentation

DijonnaWatson committed:
Update TestComputeEngineIntegration.java

fixing indentation for TestComputeEngineIntegration file

ViceRaven committed:
Update TestFactorComputationCordinator.java

updated when statements to reflect result and status mocks

ViceRaven committed:
Update TestFactorComputationCordinator.java

attempted to satisfy gradle by adding missing import statement for the computer result interface

DijonnaWatson committed:
Add files via upload

DijonnaWatson committed:
Update FactorComputeResultImpl.java

added this implementation of the computeResults Interface since the coordinator test class kept saying the getStatus method was pointing to a null pointer. I don't know if this actually helped but I did have this class when solving that problem.

DijonnaWatson committed:
Delete src/factorFinder/FactorComputeResultImpl.java

nevermind we don't need this, i checked and it still works without this implementation, it was the fact that we weren't using the mock request we made and put it into the prototype as a parameter, to then use the compute method and see if the getStatus method worked. I think that's why we were getting a null pointer

DijonnaWatson committed:
Update TestFactorComputationCordinator.java

I did change the parameters of the prototype, but this works now. We were calling the .compute() method on a request on the mock status we made. So I think that's why we got the null exception

DijonnaWatson committed:
Update TestFactorComputationCordinator.java

trying to fix the "import factorfinder.FactorComputeResult.ComputeResultStatus;"
import

ViceRaven committed:
Update TestFactorComputationCordinator.java

fixed wrong import statement

ViceRaven committed:
Update TestFactorComputationCordinator.java

attempting to fix gradle by adding missing import statements

ViceRaven committed:
Update TestFactorComputationCordinator.java

removed unneeded import

ViceRaven committed:
Update TestFactorComputationCordinator.java

removedd factorcomputeresultimpl because it does not exist

ViceRaven committed:
Update TestFactorComputationCordinator.java

attempted to fix gradle by adding missing imports

DijonnaWatson commented on test/TestFactorComputationCordinator.java:
```diff
@@ -11,79 +16,48 @@ public void testFactorComputationCordinator() {
 
     // Mock the result
-    FactorComputeResult resultMock = Mockito.mock(FactorComputeResult.class);
+    FactorComputeResult resultMock =
+        Mockito.mock(FactorComputeResult.class);
```
Thanks for fixing that, I meant to update that line but forgot

DijonnaWatson commented on src/factorFinder/FactorComputeEngineImpl.java:
```diff
@@ -0,0 +1,44 @@
+        factors.add(i); // Add factor to list
+      }
+    }
+    // Convert ArrayList to array
+    //.mapToInt() operation converts each Integer object in the stream to its
+    // corresponding int primitive value. (Integer::intValue) is shortHand for
+    // the intValue() method one EACH Integer Object in the stream. Integer ->
+    // Integer.intValue() is what it should look like when expanded .toArray
+    // collects the elements in the IntStream and returns them as an int[]
+    // array.
```
Do you guys think my comment for this line helps you understand it, or is this too much?

ViceRaven replied:
Yeah it works

jeremy23newman replied:
I think it helps with readability

DijonnaWatson commented on test/TestFactorComputationCordinator.java:
```diff
@@ -1,8 +1,13 @@
-
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
+import com.example.factorfinder.FactorComputeResult.ComputeResultStatus;
+import com.example.factorfinder.FactorComputeResult;
+import com.example.factorfinder.FactorComputeRequest;
+import com.example.factorfinder.FactorComputationCordinator;
+
```
Good job on fixing the imports

DijonnaWatson commented on test/TestFactorComputationCordinator.java:
```diff
@@ -1,8 +1,13 @@
-
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
+import com.example.factorfinder.FactorComputeResult.ComputeResultStatus;
+import com.example.factorfinder.FactorComputeResult;
+import com.example.factorfinder.FactorComputeRequest;
+import com.example.factorfinder.FactorComputationCordinator;
```
Good job o fixing the imports

DijonnaWatson commented on src/factorFinder/FactorComputationCoordinatorImpl.java:
```diff
@@ -0,0 +1,66 @@
+  @Override
+  public FactorComputeResult compute(FactorComputeRequest request) {
+    // Correlates to step 2a. Here we receive requests from the user to start the computation
+    InputConfig inputConfig =
+        request.getInputConfig(); 
+
```
This is still using the interface instead of the implementation of the InputConfig (FileInputConFig), I'm fixing it right now  in eclipse to see if anything will be messed up

DijonnaWatson commented on src/factorFinder/FactorComputationCoordinatorImpl.java:
```diff
@@ -0,0 +1,66 @@
+          computeEngine.getFactors(number); // Compute the passed integer's factors
+
+      // Correlates to step 2d. Request that the data storage component write whatever result to the output
+      OutputConfig outputConfig =
+          request.getOutputConfig(); 
+      if (outputConfig != null) {
```
same for this one, it's using the interface and not the FileOutputConfig

DijonnaWatson replied:
ill hyave to update the factorComputeRequest class too

DijonnaWatson replied:
and the FactorComputationCordinatorPrototype, I think the test is still fine since I was already using the implementations of the input and output configs.

DijonnaWatson committed:
Update TestComputeEngineIntegration.java

correcting the implementation of FileOutputConfig

DijonnaWatson committed:
Update InMemoryFactorDataStore.java

correcting the instances of the FileInputConfig and FileOutputConfig

DijonnaWatson committed:
Update FactorDataStorePrototype.java

we're not really using this anymore but I updated the use of FileOutputConfig and FileInputConfig

DijonnaWatson committed:
Update FactorDataStoreImpl.java

updating the use of FileInputConfig and FileOutputonfig

DijonnaWatson committed:
Update FactorDataStorePrototype.java

indentation

DijonnaWatson committed:
Update FactorDataStorePrototype.java

fixing checkstyle

DijonnaWatson committed:
Update FactorDataStore.java

changing InputConfig and outPutConfig to FIleInputConfig and FileOutputConfig

DijonnaWatson committed:
Update FactorComputeRequest.java

upating to use FileInputConfig and FileOutputConfig

DijonnaWatson committed:
Update FactorComputationCordinatorPrototype.java

updating to use the FIle input and FIle output configs

DijonnaWatson committed:
Update FactorComputationCoordinatorImpl.java

updating so it uses the file input and file output configs

DijonnaWatson committed:
Update FactorComputeRequest.java

fixing package

ViceRaven committed:
Update FactorComputeRequest.java

fixed import statements

DijonnaWatson committed:
Update FactorDataStorePrototype.java

fixing indentation

ViceRaven committed:
Update FactorComputationCoordinatorImpl.java

fixed misspelling of coordinator

ViceRaven committed:
Update FactorComputationCoordinatorImpl.java

fixed misspelling of coordinator

ViceRaven committed:
Update FactorComputationCoordinatorImpl.java

fixed misspelling

DijonnaWatson committed:
Update FactorDataStorePrototype.java

fixing indentation

ViceRaven committed:
Update TestFactorDataStore.java

fixed line 30

ViceRaven commented on src/factorFinder/FactorComputationCordinatorPrototype.java:
```diff
@@ -1,57 +1,27 @@
 
+import java.util.List;
+
 public class FactorComputationCordinatorPrototype {
```
At some point we need to fix the spelling of coordinator. There were a lot of issues that were resolved by just changing the spelling of coordinator.

jeremy23newman commented on src/factorFinder/FactorComputeEngineImpl.java:
```diff
@@ -0,0 +1,44 @@
+      }
+    }
+    return formattedOutput.toString();
+  }
```
looks good to me

jeremy23newman commented on src/factorFinder/FactorDataStoreImpl.java:
```diff
@@ -0,0 +1,54 @@
+      }
+    } catch (IOException e) {
+      System.err.println("Error reading from file: " + e.getMessage());
+    }
```
good job with the try and catch block, and the error being thrown

jeremy23newman commented on test/TestComputeEngineIntegration.java:
```diff
@@ -2,24 +2,30 @@
+import java.util.stream.IntStream;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
```
i believe that all of these imports are necessary

jeremy23newman commented on test/TestFactorComputationCordinator.java:
```diff
@@ -11,79 +16,48 @@ public void testFactorComputationCordinator() {
+      FactorComputationCordinator apiToCall, FactorComputeRequest request) {
+    // Call the compute method using the apiToCall instance with the mocked
+    // request
     FactorComputeResult result = apiToCall.compute(request);
```
nice comments, helps understand the code better

jeremy23newman committed:
adding two additional tests

DijonnaWatson reviewed (approved):
i think everything looks good.

ViceRaven committed:
Update InMemoryInputConfig.java
